src/prefect/experimental/workers/base.py

```diff
@@ -560,9 +584,9 @@ def get_status(self):
 
     async def _get_configuration(self, flow_run: FlowRun) -> BaseJobConfiguration:
         deployment = await self._client.read_deployment(flow_run.deployment_id)
-        configuration = self.job_configuration.from_template_and_overrides(
+        configuration = await self.job_configuration.from_template_and_values(
             base_job_template=self._work_pool.base_job_template,
-            deployment_overrides=deployment.infra_overrides,
+            values=deployment.infra_overrides,
```

Comment:
```suggestion
            values=deployment.infra_overrides or {},
```
not sure if this is a nullable field, so could be worth hedging against